src/test/java/org/broadinstitute/hellbender/tools/walkers/annotator/VariantAnnotatorIntegrationTest.java

```diff
@@ -0,0 +1,392 @@
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.*;
+import java.util.function.BiConsumer;
+import java.util.stream.Collectors;
+
+public class VariantAnnotatorIntegrationTest extends CommandLineProgramTest {
+    final static String STANDARD_ANNOTATIONS = " -G StandardAnnotation ";
+    private static final List<String> ATTRIBUTES_TO_IGNORE = Arrays.asList(
+            "QD",//TODO QD has a cap value and anything that reaches that is randomized.  It's difficult to reproduce the same random numbers across gatk3 -> 4
+            "FS");//TODO There's some bug in either gatk3 or gatk4 fisherstrand that's making them not agree still, I'm not sure which is correct
+    private static final List<String> HEADER_LINES_TO_IGNORE = Arrays.asList(
+            "reference",
+            "HaplotypeScore",
+            "GATKCommandLine.VariantAnnotator",
+            "RAW_MQ"
+    );
+
+    //TODO because of differences between how GATK3 and GATK4 handle capturing reads for spanning deletions (Namely 3 only looks for reads overlapping the first site, 4 gets all reads over the span)
+    //TODO then we want ot ignore affected attributes for concordance tests
+    private static final List<String> DEPTH_ATTRIBUTES_TO_IGNORE = Arrays.asList(
+            "QD",
+            "DP",
+            "MQ");
+
+    final String testFile = getToolTestDataDir() + "vcfexample2.vcf";
+
+    public static String baseTestString() {
+        return "-R " + b37_reference_20_21 + " -O %s ";
+    }
+
+    //==================================================================================================================
+    // Testing
+    //==================================================================================================================
+    private void assertVariantContextsMatch(File input, File expected, List<String> extraArgs, String reference) throws IOException {
+        assertVariantContextsMatch(input, expected, extraArgs,reference, ATTRIBUTES_TO_IGNORE);
+    }
+
+    private void assertVariantContextsMatch(File input, File expected, List<String> extraArgs, String reference, List<String> ignoreAtrributes) throws IOException {
+        final VCFHeader header = getHeaderFromFile(expected);
+
+        runVariantAnnotatorAndAssertSomething(input, expected, extraArgs, (a, e) -> {
+            VariantContextTestUtils.assertVariantContextsAreEqualAlleleOrderIndependent(a, e, ATTRIBUTES_TO_IGNORE, header);
+        }, reference);
+    }
+
+    //TODO if this gets generalized to another class this should be made more robust to differing number of header lines
+    private void assertHeadersMatch(File input, File expected, List<String> linesToIgnore) throws IOException {
+        final VCFHeader expectedHeader = getHeaderFromFile(expected);
+        final VCFHeader inputHeader = getHeaderFromFile(input);
+
+        Iterator<VCFHeaderLine> inputItr = inputHeader.getMetaDataInSortedOrder().iterator();
```

Comment:
I think this could be simplified by like:
```
        final Iterator<VCFHeaderLine> itr = expectedHeader.getMetaDataInSortedOrder().stream().filter(line -> !shouldFilterLine(line, linesToIgnore)).iterator();
        final Iterator<VCFHeaderLine> itr2 = inputHeader.getMetaDataInSortedOrder().stream().filter(line -> !shouldFilterLine(line, linesToIgnore)).iterator();
        Assert.assertEquals(itr, itr2);
```